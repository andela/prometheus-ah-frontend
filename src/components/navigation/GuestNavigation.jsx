import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import ModalForm from '../modals/Modal';


/**
 * @description class for userNavigation header
 *
 * @class GuestNavigation
 *
 * @extends {Component}
 */
class GuestNavigation extends Component {
  /**
   * @description Render the JSX template
   *
   * @returns {JSX} render JSX template
   *
   * @memberof Header
   */
  render() {
    const { showLoginModal, showSignupModal, modal } = this.props;
    return (
      <nav className="navbar-nav">
        <li className="nav-item">
          <a className="nav-link" href="/">Home</a>
        </li>
        <li className="nav-item">
          <Link
            to="#"
            className="nav-link"
            onClick={showLoginModal}
          >
            Login
          </Link>
          <ModalForm modal={modal} />
        </li>
        <li className="nav-item" id="signup">
          <Link
            to="#"
            className="nav-link"
            onClick={showSignupModal}
          >
            Sign Up
          </Link>
          <ModalForm modal={modal} />
        </li>
      </nav>
    );
  }
}

GuestNavigation.propTypes = {
  showLoginModal: PropTypes.func,
  showSignupModal: PropTypes.func,
  modal: PropTypes.shape({})
};

export default GuestNavigation;
