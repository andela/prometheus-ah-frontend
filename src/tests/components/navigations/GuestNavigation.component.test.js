import React from 'react';
import { shallow } from 'enzyme';
import GuestNavigation from '../../../components/navigation/GuestNavigation';

let wrapper;

describe('GuestNavigation component', () => {
  beforeEach(() => {
    const props = {
      showLoginModal: jest.fn(),
      showSignupModal: jest.fn(),
      modal: {},
    };
    wrapper = shallow(<GuestNavigation {...props} />);
  });

  it('should renders a snapshot of the guestNavigation component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
