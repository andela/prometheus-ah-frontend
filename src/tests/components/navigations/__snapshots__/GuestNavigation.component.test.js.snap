// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestNavigation component should renders a snapshot of the guestNavigation component 1`] = `
<ul
  className="navbar-nav ml-auto"
>
  <li
    className="nav-item mr-4"
  >
    <a
      className="nav-link"
      href="/"
    >
      Home
    </a>
  </li>
  <form
    className="form-inline mr-3"
  >
    <i
      className="mdi mdi-magnify mdi-24px mr-2 d-none d-md-block"
    />
    <input
      aria-label="Search"
      className="form-control mr-sm-2 border-0"
      placeholder="Search..."
      type="search"
    />
  </form>
  <li
    className="nav-item nav-btn mr-4"
    id="login"
  >
    <Link
      className="nav-link login-link"
      onClick={[MockFunction]}
      replace={false}
      to="#"
    >
      Login
    </Link>
    <Connect(Modal)
      modal={Object {}}
    />
  </li>
  <li
    className="nav-item nav-btn"
    id="signup"
  >
    <Link
      className="nav-link register-link"
      onClick={[MockFunction]}
      replace={false}
      to="#"
    >
      Register
    </Link>
    <Connect(Modal)
      modal={Object {}}
    />
  </li>
</ul>
`;
